app/controllers/users_controller.rb

```diff
@@ -0,0 +1,39 @@
+class UsersController < ApplicationController
+  before_action :set_user, only: %i[show edit update destroy]
+  def index
+    @users = User.all
+  end
+
+  def show; end
+
+  def edit; end
+
+  def update
+    @user = current_user
```

Comment:
I would not do a kind of authorization here. I would set the user with the id from the request.

Reply:
agree, done ✅